ethcore/engines/authority-round/src/lib.rs

```diff
@@ -437,8 +438,11 @@ impl EpochManager {
 /// proof.
 #[derive(Clone, Debug, PartialEq, Eq)]
 struct EmptyStep {
+	/// The signature of the other two fields, by the message's author.
 	signature: H520,
+	/// This message's step number.
 	step: u64,
+	/// The hash of the most recent block.
```

Comment:
Is this the hash of the parent to the block where the authority emits the `EmptyStep` message?

Reply:
If I understand correctly, an empty step message is created instead of a block, if a validator has no pending transactions. It cannot itself be a parent, and this hash always points to the most recent block.
E.g.:
* Validator A creates block 1.
* Validator B has no pending transaction, so it signs an empty step message m<sub>B</sub> instead whose hash points to block 1.
* Validator C also has no pending transactions, so it also signs an empty step message m<sub>C</sub> instead whose hash points to block 1.
* Validator D creates block 2. The parent is block 1, and the header includes m<sub>B</sub> and m<sub>C</sub>.

(Would be good if someone could confirm my understanding!)

Reply:
Oh right, makes sense now. Thank you for explaining.

Reply:
Maybe your excellent comment here could go into the module level docs or the docs for `EmptyStep` for the benefit of the next clueless code reader? :)

Reply:
Done.